app/models/carriage.rb

```diff
@@ -1,6 +1,30 @@
 # frozen_string_literal: true
 
 class Carriage < ApplicationRecord
-  validates :carriage_type, presence: true
   belongs_to :train
+
+  validates :number, presence: true
+
+  scope :economy, -> { where(type: 'EconomyCarriage') }
+  scope :coupe, -> { where(type: 'CoupeCarriage') }
+  scope :sv, -> { where(type: 'SVCarriage') }
+  scope :seated, -> { where(type: 'SeatedCarriage') }
+
+  after_create :add_number_in_train
+
+  def self.carriage_types
+    %w[EconomyCarriage CoupeCarriage SVCarriage SeatedCarriage]
+  end
+
+  def add_number_in_train
+    carriages_in_train = Carriage.where(train_id: train_id)
+    number_in_train_presented = carriages_in_train.select { |carriage| carriage.number_in_train.present? }
+
+    self.number_in_train = if number_in_train_presented.present?
```

Comment:
where do you find this life hook with self?

Reply:
Is thats wrong? Should I rewrite with update(number_in_train: )?